lib/surface/compiler.ex

```diff
@@ -144,10 +144,22 @@ defmodule Surface.Compiler do
     num_tags =
       ast
       |> Enum.filter(fn
-        %AST.Tag{} -> true
-        %AST.VoidTag{} -> true
-        %AST.Component{} -> true
-        _ -> false
+        %AST.Tag{} ->
+          true
+
+        %AST.VoidTag{} ->
+          true
+
+        %AST.Component{type: Surface.LiveComponent, meta: meta} ->
+          warn_live_component_as_root_node_of_another_live_component(meta, caller, offset)
+
+          false
```

Comment:
Shouldn't be `true`? 

For the most common scenario for this warning, we'll have a single LiveComponent as root. By returning `false`, the `num_tags` will be `0`, resulting in two warnings. The one we expected and another `stateful live components must have a HTML root element`, which can be confusing. WDYT?

**Note:** it might be a good idea to `assert` against the whole `output` using `==` instead of `=~`. This way we make sure we don't have more warnings than we expect.

Reply:
I posted the same comment few minutes ago ^^ https://github.com/surface-ui/surface/pull/290#discussion_r593397013. I agree, it should be true.

I'll assert the warning with `==` but to be consistent we should update the code everywhere we use `=~` to test warnings. If you think it's important, I can take some time this weekend to update the tests.

Reply:
Haha. Sorry, I didn't see your comment :)

> we should update the code everywhere we use =~ to test warnings. If you think it's important, I can take some time this weekend to update the tests.

On second thought, forget about using `==`. It's not worth it. Just use `=~` as before and let's move on. I'll think of a better way to assert we don't have extra warnings.

Feel free to merge it after this change. Great work! ❤️

Reply:
The tests do not pass on the CI  but they pass on my computer because my terminal handles colours. The output of the warning on my computer starts with `\e[33mwarning: \e[0m` but `warning:` in the CI. It's maybe for that reason why we use `= ~` instead of `==`. I'll look a this tommorow.

Reply:
Too many cross messages here :sweat_smile: